job_creator.php

```diff
@@ -375,7 +377,7 @@ private function getListOfPhpVersionsByBranchName(): array
         $json = $this->getComposerJsonContent();
         if ($json) {
             $php = $json->require->php ?? null;
-            $php = str_replace('^', '', $php);
+            $php = str_replace('^', '', $php ?? '');
```

Comment:
Fixes deprecation notice in php 8.3

Deprecated: str_replace(): Passing null to parameter #3 ($subject) of type array|string is deprecated in /home/sboyd/Modules/gha-generate-matrix/job_creator.php on line 378